### ISSUE TITLE
Compatible with 0.6.8?

### ISSUE THREAD

**Issue by sdodson** [vpavlin/docker-registry-gc#1]
Any idea if this is compatible with docker-registry-0.6.8-8 as available in rhel/centos 7? I ran this and now I cannot push a newly created tag, though pushing to existing tags seems to work. I'm not certain it's related to running this tool but this is the backtrace

 gunicorn[14214]: 2015-02-27 17:00:45,814 ERROR: Exception on /v1/repositories/openshift3_beta/ose-pod/ [PUT]
 gunicorn[14214]: Traceback (most recent call last):
 gunicorn[14214]: File "/usr/lib/python2.7/site-packages/flask/app.py", line 1817, in wsgi_app
 gunicorn[14214]: response = self.full_dispatch_request()
 gunicorn[14214]: File "/usr/lib/python2.7/site-packages/flask/app.py", line 1477, in full_dispatch_request
 gunicorn[14214]: rv = self.handle_user_exception(e)
 gunicorn[14214]: File "/usr/lib/python2.7/site-packages/flask/app.py", line 1381, in handle_user_exception
 gunicorn[14214]: reraise(exc_type, exc_value, tb)
 gunicorn[14214]: File "/usr/lib/python2.7/site-packages/flask/app.py", line 1475, in full_dispatch_request
 gunicorn[14214]: rv = self.dispatch_request()
 gunicorn[14214]: File "/usr/lib/python2.7/site-packages/flask/app.py", line 1461, in dispatch_request
 gunicorn[14214]: return self.view_functions[rule.endpoint](**req.view_args)
 gunicorn[14214]: File "/usr/lib/python2.7/site-packages/docker-registry/docker_registry/toolkit.py", line 267, in wrapper
 gunicorn[14214]: return f(namespace, repository, _args, *_kwargs)
 gunicorn[14214]: File "/usr/lib/python2.7/site-packages/docker-registry/docker_registry/toolkit.py", line 241, in wrapper
 gunicorn[14214]: return f(_args, *_kwargs)
 gunicorn[14214]: File "/usr/lib/python2.7/site-packages/docker-registry/docker_registry/index.py", line 105, in put_repository
 gunicorn[14214]: update_index_images(namespace, repository, flask.request.data)
 gunicorn[14214]: File "/usr/lib/python2.7/site-packages/docker-registry/docker_registry/index.py", line 72, in update_index_images
 gunicorn[14214]: iid = i['id']
 gunicorn[14214]: KeyError: 'id'

**Comment by vpavlin** [comment]
Hmm, I am not sure and I will not have access to my laptop till Monday.. But exactly for these occasions it backs up _index_images so you can replace generated ones

**Comment by vpavlin** [comment]
Hi, it's fixed by e7b546372300b685cc167f1298f407f6f9e8754d - it was bug and broke all registry versions..

You can easily "repair" it by running

```
for i in `grep -rl Id /var/lib/docker-registry/repositories/*`; do sed -i 's/"Id"/"id"/g' $i; done
```

Sorry for this inconvenience.

**Comment by sdodson** [comment]
Vaclav, Thanks!
